src/v/resource_mgmt/CMakeLists.txt

```diff
@@ -13,6 +13,7 @@ v_cc_library(
     storage.cc
   DEPS
     Seastar::seastar
+    v::cloud_storage
```

Comment:
nit: this is interesting. Note to self/storage, cloud_storage/cache_service doesn't have any dependencies on anything else in cloud_storage. Maybe it belongs somewhere else (though I don't feel strongly, and can't think of a better home at the moment)

Reply:
Do you mean the control loop may need to belong somewhere else?

Reply:
No, I'm just noticing that this module probably doesn't need to conceptually depend on cloud_storage (manifests, remote storage calls, etc) since it deals entirely with local state. It's not problematic right now, but stood out as an initially surprising dependency (I briefly forgot the cache was in v::cloud_storage).

Reply:
oh, if i'm reading this correctly the second time, you're suggesting that the cache service could be its own library? yeh, i think that would make sense.

in fact, when you look at the issue solved by this cmakelists.txt change, it's because we need a transitive dependency on hashing library. this could even be hidden inside a cache service compilation unit rather than making the header be an issue:

```
-error=#warnings" -fstack-clash-protection -fsanitize=address -fsanitize=undefined -fno-sanitize=vptr -MD -MT src/v/resource_mgmt/CMakeFiles/v_storage_resource_mgmt.dir/storage.cc.o -MF src/v/resource_mgmt/CMakeFiles/v_storage_resource_mgmt.dir/storage.cc.o.d -o src/v/resource_mgmt/CMakeFiles/v_storage_resource_mgmt.dir/storage.cc.o -c /home/nwatkins/src/redpanda/src/v/resource_mgmt/storage.cc
In file included from /home/nwatkins/src/redpanda/src/v/resource_mgmt/storage.cc:14:
In file included from /home/nwatkins/src/redpanda/src/v/cloud_storage/cache_service.h:13:
In file included from /home/nwatkins/src/redpanda/src/v/cloud_storage/access_time_tracker.h:14:
/home/nwatkins/src/redpanda/src/v/hashing/xx.h:74:19: error: field has incomplete type 'XXH64_state_t' (aka 'XXH64_state_s')
    XXH64_state_t _state{};
                  ^
/home/nwatkins/src/redpanda/vbuild/debug/clang/rp_deps_install/include/xxhash.h:717:16: note: forward declaration of 'XXH64_state_s'
typedef struct XXH64_state_s XXH64_state_t;   /* incomplete type */
```

Reply:
Right, exactly. Interesting point about the transitive dep.

Anyway, I don't feel strongly enough to gate this PR on changes here, just food for thought